tests/transport/register.c

```diff
@@ -44,6 +44,8 @@ void test_transport_register__custom_transport_ssh(void)
 
 #ifndef GIT_SSH
 	cl_git_fail_with(git_transport_new(&transport, NULL, "ssh://somehost:somepath"), -1);
+	cl_git_fail_with(git_transport_new(&transport, NULL, "ssh+git://somehost:somepath"), -1);
+	cl_git_fail_with(git_transport_new(&transport, NULL, "git+ssh://somehost:somepath"), -1);
```

Comment:
It looks like you're only testing that `ssh+git` fails when not compiling with `GIT_SSH`.  But I would expect that with or without this patch.

Can you test that you can successfully build a transport with `git+ssh` and with `ssh+git` below in the else block?